### PR TITLE
feat: Add Qiacuity dPCR adapter

### PR DESCRIPTION
**Pull request by dchou-benchling**
See requirements document here: https://docs.google.com/document/d/17kQ-qfbsb4iuss5Lk7hhxxAVvzkoMeOs4W5R6Gkr-r0/edit#heading=h.cgvpd23o8pf0

### CODE REVIEW

File: src/allotropy/parsers/qiacuity_dpcr/qiacuity_dpcr_parser.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+        self,
+        well_item: pd.Series,
+        sample_document: SampleDocument,
+        device_control_aggregate_document: DeviceControlAggregateDocument,
+        processed_data_aggregate_document: ProcessedDataAggregateDocument,
+    ) -> MeasurementDocumentItem:
+        # Assign a random measurement id
+        measurement_id = random_uuid_str()
+        # There is no measurement time in the file, so assign to unix epoch
+        measurement_time = EPOCH
```
Comment (nathan-stender) [inline src/allotropy/parsers/qiacuity_dpcr/qiacuity_dpcr_parser.py:136]: I wonder if we could use file creation time? I know you wrote the requirements on this, so I'll defer to you if you think this is a reasonable interpretation.

File: src/allotropy/parsers/qiacuity_dpcr/qiacuity_dpcr_parser.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        )
+        sample_document = SampleDocument(sample_identifier=sample_identifier)
+
+        sample_role_type = self._get_value_by_col_name(
+            well_item, SAMPLE_TYPE_COLUMN_NAME, required=False
+        )
+        # TODO: When the sample role type model is updated in this repo, we should update this
+        # Map sample role types to valid sample role types from ASM
+        if sample_role_type is not None:
+            sample_role_type = SAMPLE_ROLE_TYPE_MAPPING[sample_role_type]
```
Comment (nathan-stender) [inline src/allotropy/parsers/qiacuity_dpcr/qiacuity_dpcr_parser.py:169]: Error handle here if `sample_role_type` is not in mapping (raise `AllotropeConversionError`)

File: src/allotropy/parsers/qiacuity_dpcr/qiacuity_dpcr_parser.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+    ) -> DataProcessingDocument:
+        return DataProcessingDocument(
+            flourescence_intensity_threshold_setting=TQuantityValueUnitless(
+                value=self._get_value_by_col_name(
+                    well_item, FIT_SETTING_COLUMN_NAME, required=False
+                )
+            )
+        )
+
+    def _get_value_by_col_name(
```
Comment (nathan-stender) [inline src/allotropy/parsers/qiacuity_dpcr/qiacuity_dpcr_parser.py:232]: Should should be able to replace this with `try_str_from_series` (when required=True) and `try_str_from_series_or_none` (when required=False)

**Review by nathan-stender — COMMENTED** [review]
Great PR!

Left a few comments for using utility functions and some misc.

Other than that, can you please add a line to `added` in the `CHANGELOG` and add this to the list of supported adapters in `README`?

For the title, we don't use BNCH tickets in this repo since it's not Benchling specific. I appriciate the detailed title though, you can prefix with `feat:`

Finally, don't forget to add yourself to the `authors` list in `pyproject.yaml` :D

**Review by nathan-stender — COMMENTED** [review]
Oh p.s. you've got some lint errors. You can run `hatch run lint:fmt` to autofix what can be fixed, and `hatch run lint:all` to show you all errors.

File: src/allotropy/parsers/qiacuity_dpcr/qiacuity_dpcr_parser.py (hunk 3 above)
Comment (dchou-benchling) [reply src/allotropy/parsers/qiacuity_dpcr/qiacuity_dpcr_parser.py:232]: Thank you! Looks like this was the cause of my lint failure

File: src/allotropy/parsers/qiacuity_dpcr/qiacuity_dpcr_parser.py (hunk 1 above)
Comment (dchou-benchling) [reply src/allotropy/parsers/qiacuity_dpcr/qiacuity_dpcr_parser.py:136]: Yeah I talked to Joe about this today, sounds like he's ok leaving it for now

File: src/allotropy/parsers/qiacuity_dpcr/qiacuity_dpcr_parser.py (hunk 3 above)
Comment (dchou-benchling) [reply src/allotropy/parsers/qiacuity_dpcr/qiacuity_dpcr_parser.py]: This is updated, resolving the conversation

File: src/allotropy/parsers/qiacuity_dpcr/qiacuity_dpcr_parser.py (hunk 2 above)
Comment (dchou-benchling) [reply src/allotropy/parsers/qiacuity_dpcr/qiacuity_dpcr_parser.py]: This is updated!

**Pull request merged**